telluric/georaster.py

```diff
@@ -95,20 +101,56 @@ def merge_all(rasters, roi=None, dest_resolution=None, merge_strategy=MergeStrat
 
     # Create a list of single band rasters
     all_band_names, projected_rasters = _prepare_rasters(rasters, merge_strategy, empty)
+    assert len(projected_rasters) == len(rasters)
+
+    prepared_rasters = _apply_pixel_strategy(projected_rasters, pixel_strategy)
+
+    # Extend the rasters list with only those that have the requested bands
+    prepared_rasters = _explode_rasters(prepared_rasters, all_band_names)
 
     if all_band_names:
         # Merge common bands
-        projected_rasters = _merge_common_bands(projected_rasters)
+        prepared_rasters = _merge_common_bands(prepared_rasters)
 
         # Merge all bands
-        raster = reduce(_stack_bands, projected_rasters)
+        raster = reduce(_stack_bands, prepared_rasters)
 
         return empty.copy_with(image=raster.image, band_names=raster.band_names)
 
     else:
         raise ValueError("result contains no bands, use another merge strategy")
 
 
+def _apply_pixel_strategy(rasters, pixel_strategy):
+    # type: (List[Optional[_Raster]], PixelStrategy) -> List[_Raster]
+    if pixel_strategy == PixelStrategy.METADATA:
+        new_rasters = []
+        for ii, raster in enumerate(rasters):
+            if raster:
+                new_image = np.ma.masked_array(
+                    np.full_like(raster.image.data, ii, dtype=int),
+                    raster.image.mask
+                )
+                new_rasters.append(_Raster(image=new_image, band_names=raster.band_names))
+
+        return new_rasters
+
+    else:
+        # The way merge_all is written now, this pixel strategy is the default one
```

Comment:
don't you think that is better to do an `if` here to check the enum , and to throw exception of the strategy is not known?

Reply:
I considered it, yes - will change that